Added an option to use the pure python implementation. (#1137)
tensorflow/addons: README.md

conform to established API patterns seen in core TensorFlow.

#### GPU/CPU Custom-Ops
A major benefit of TensorFlow Addons is that there are precompiled ops.
Custom ops work on GPU only on Ubuntu. This is why on Windows and MacOS, 

> If available, Addons will fallback to a pure TensorFlow Python implementation for custom-ops.

Reviewer: ```suggestion
A major benefit of TensorFlow Addons is that there are precompiled ops for CPU/GPU.
Currently however, GPU custom ops only work for Linux distributions. For this reason Windows and MacOS will fallback to pure TensorFlow Python implementations whenever possible.
```

The order of priority in MacOS/Windows:
1) Pure TensorFlow + Python implementation (work on cpu+gpu)
2) C++ implementation for CPU

> The order of priority for Ubuntu:

Reviewer: ```suggestion
The order of priority for Linux:
```

1) CUDA implementation
2) C++ implementation
3) Pure TensorFlow + Python implementation (work on cpu+gpu)

If you want to change the default priority, "C++ and CUDA" VS "pure TF Python", 
you can either set the variable `TF_ADDONS_PY_OPS` from the command line or in 
your code.

For example, if you're on linux and you have compatibility problems with the compiled ops,
and you want to give priority to the Python implementation
you can do:

From the command line:
```
export TF_ADDONS_PY_OPS=1
```

or in your code:

```
import tensorflow_addons as tfa
tfa.options.TF_ADDONS_PY_OPS=True
```

> This variable default to `True` on Windows and Mac, and `False` for Linux.

Reviewer: ```suggestion
This variable will default to `True` on Windows and Mac, and `False` for Linux.
```
